automation/docs/generate-javadocs.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+set -euo pipefail
+
+for module in $JVM_PROJECTS
+ do
+   echo "Generating javadocs for module: $module and all it's sub-modules"
+   mvn -f "$module"/pom.xml javadoc:javadoc
+
+   echo "Move generated javadocs to respective doc site location"
+
+   echo "getting locations where javadocs got generated"
+   generated_doc_paths=($(find "$module" -type d  -path '*/target/site/apidocs' | sed -r 's|/[^/]+$||' | sed -r 's|/[^/]+$||' | sed -r 's|/[^/]+$||'))
```

Comment:
This is problematic. You should probably use  `mapfile` since this returns multiple lines.

Something like:

```
mapfile -t generated_doc_paths < <(find "$module" -type d  -path '*/target/site/apidocs' | sed -r 's|/[^/]+$||' | sed -r 's|/[^/]+$||' | sed -r 's|/[^/]+$||')
```
This should help:
https://github.com/koalaman/shellcheck/wiki/SC2207


Adjust and test please.